include/libdnf/transaction/comps_group.hpp

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright (C) 2017-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef LIBDNF_TRANSACTION_COMPS_GROUP_HPP
+#define LIBDNF_TRANSACTION_COMPS_GROUP_HPP
+
+
+#include "transaction_item.hpp"
+
+#include <memory>
+#include <vector>
+
+
+namespace libdnf::transaction {
+
+
+class CompsGroupPackage;
+class Transaction;
+
+
+enum class CompsPackageType : int {
+    CONDITIONAL = 1 << 0,  // a weak dependency
+    DEFAULT = 1 << 1,      // installed by default, but can be unchecked in the UI
+    MANDATORY = 1 << 2,    // always installed
+    OPTIONAL = 1 << 3      // not installed by default, but can be checked in the UI
+};
```

Comment:
Not sure whether the description has the right effect. Even mandatory doesn't mean must be installed.

Reply:
I've added a comment explaining how comps work.